src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_1_TypicalWorkflow
+    {
+        [Workflow(runMethod: nameof(SayHelloAsync))]
+        public class SayHelloWorkflow
+        {
+            private const string AddresseeNameDefault = "Boss";
+
+            private string _addresseeName = null;
+
+            public async Task SayHelloAsync(WorkflowContext workflowCtx)
+            {
+                var greeting = new SpeechRequest($"Hello, {_addresseeName ?? AddresseeNameDefault}.");                
+                await workflowCtx.Orchestrator.Activities.ExecuteAsync("SpeakAGreeting", greeting);
+            }
+
+            [WorkflowSignalHandler]
+            public Task SetAddresseeAsync(SpeechRequest input, WorkflowContext _)
+            {
+                _addresseeName = input?.Text;
+                return Task.CompletedTask;                
```

Comment:
Can't this be omitted in an async function w/ Task return type?

Reply:
is you use the async keyword - yes. But then you'll generate the entire state machine instance.